support-frontend/assets/helpers/contributions.ts

```diff
@@ -134,15 +134,35 @@ const getAmount = (
 	selectedAmounts: SelectedAmounts,
 	otherAmounts: OtherAmounts,
 	contributionType: ContributionType,
+	coverTransactionCostSelected?: boolean,
 ): number => {
 	const selectedAmount = selectedAmounts[contributionType];
 	const otherAmount = otherAmounts[contributionType].amount ?? '';
 
-	return parseFloat(
-		selectedAmount === 'other' ? otherAmount : selectedAmount.toString(),
+	// Only cover transaction costs for one off contributions
+	const coverTransactionCost =
+		coverTransactionCostSelected && contributionType === 'ONE_OFF';
+
+	return getAmountWithTransactionCostCovered(
+		parseFloat(
+			selectedAmount === 'other' ? otherAmount : selectedAmount.toString(),
+		),
+		coverTransactionCost,
 	);
 };
 
+function getAmountWithTransactionCostCovered(
```

Comment:
I'm not sure this method makes things more clear - but it doesn't make things less, which could be an indication it could just be inline.